app/models/spree_amazon/order.rb

```diff
@@ -44,8 +44,16 @@ def close_order_reference!
     end
   end
 
-  def save_total
-    mws.set_order_data(total, currency)
+  def set_order_reference_details(total, options={})
+    # These are parameters that the Amazon gem accepts
+    options.assert_valid_keys(
```

Comment:
Do we need to call `assert_valid_keys` both here and on  `amazon_mws.rb`? Also I'm wondering if asserting that the options hash has valid keys should be the responsibility of the amazon gem and not ours?